code/scripting/api/libs/ui.cpp

```diff
@@ -1036,6 +1038,71 @@ ADE_FUNC(getFictionMusicName, l_UserInterface_FictionViewer, nullptr,
 	return ade_set_args(L, "s", common_music_get_filename(SCORE_FICTION_VIEWER).c_str());
 }
 
+//**********SUBLIBRARY: UserInterface/ShipWepSelect
+ADE_LIB_DERIV(l_UserInterface_ShipWepSelect,
+	"ShipWepSelect",
+	nullptr,
+	"API for accessing data related to the ship and weapon select UIs.<br><b>Warning:</b> This is an internal "
+	"API for the new UI system. This should not be used by other code and may be removed in the future!",
+	l_UserInterface);
+
+ADE_FUNC(initSelect,
+	l_UserInterface_ShipWepSelect,
+	nullptr,
+	"Initializes selection data including wing slots, ship and weapon pool, and loadout information",
+	nullptr,
+	nullptr)
+{
+	//Note this does all the things from common_select_init() in missionscreencommon.cpp except load UI
+	//elements into memory - Mjn
+	
+	SCP_UNUSED(L); // unused parameter
+
+	Common_team = 0;
+
+	if ((Game_mode & GM_MULTIPLAYER) && IS_MISSION_MULTI_TEAMS)
+		Common_team = Net_player->p_info.team;
+
+	common_set_team_pointers(Common_team);
+
+	ship_select_common_init();
+	weapon_select_common_init();
+
+	if ( Game_mode & GM_MULTIPLAYER ) {
+		multi_ts_common_init();
+	}
+
+	// restore loadout from Player_loadout if this is the same mission as the one previously played
+	if ( !(Game_mode & GM_MULTIPLAYER) ) {
+		if ( !stricmp(Player_loadout.filename, Game_current_mission_filename) ) {
+			wss_maybe_restore_loadout();
+			ss_synch_interface();
+			wl_synch_interface();
+		}
+	}
+
+	return ADE_RETURN_NIL;
+}
+
+ADE_LIB_DERIV(l_Loadout_Wings, "Loadout_Wings", nullptr, nullptr, l_UserInterface_ShipWepSelect);
+ADE_INDEXER(l_Loadout_Wings, //
+	"number Index",
+	"Array of loadout wing data",
+	"loadout_wing",
+	"loadout handle, or invalid handle if index is invalid")
+{
+	int idx;
+	if (!ade_get_args(L, "*i", &idx))
+		return ade_set_error(L, "s", "");
```

Comment:
Unlike what the docs say, you're not returning an invalid handle (which'd be ``ade_set_error(L,"o", l_Loadout_Wing.Set(ss_wing_info_h()));``), but an empty string

Reply:
Fixed

Reply:
I'm reasonably sure that you still have to _return_ the ``ade_set_error`` there, not just call it.
But yeah, that'll be doing a proper invalid handle

Reply:
Whoops. Didn't intend to replace the return. Fixed